insights/parsers/jboss_version.py

```diff
@@ -49,27 +63,78 @@ def parse_content(self, content):
             "release": int(release)
         }
 
+    @property
+    def product(self):
+        """string: the version of this running JBoss progress."""
+        return self._parsed["product"]
+
     @property
     def version(self):
         """string: the version of this running JBoss progress."""
-        return self.parsed["version"]
+        return self._parsed["version"]
 
     @property
     def major(self):
         """int: the major version of this running JBoss progress."""
-        return self.parsed["major"]
+        return self._parsed["major"]
 
     @property
     def minor(self):
         """int: the minor version of this running JBoss progress."""
-        return self.parsed["minor"]
+        return self._parsed["minor"]
 
     @property
     def release(self):
         """int: release of this running JBoss progress."""
-        return self.parsed["release"]
+        return self._parsed["release"]
 
     @property
     def code_name(self):
         """string: code name of this running JBoss progress."""
-        return self.parsed["code_name"]
+        return self._parsed["code_name"]
+
+
+@parser(Specs.jboss_runtime_versions)
+class JbossRuntimeVersions(Parser):
+    """
+     This class is to access to file ``data/insights_commands/jboss_versions``
+
+     Typical content of file ``data/insights_commands/jboss_versions`` is::
+
+         {"/opt/jboss-datagrid-7.3.0-server": "Red Hat Data Grid - Version 7.3.0"}
+
+     This class parses the file content and stores data in the dict ``self.versions``.
+
+     Examples:
+         >>> len(all_jboss_versions)
+         1
+         >>> all_jboss_versions[0].major
+         7
+         >>> all_jboss_versions[0].minor
+         3
+         >>> all_jboss_versions[0].release
+         0
+     """
+
+    def parse_content(self, content):
+        self._versions = []
+        jboss_version_dict = json.loads(' '.join(content))
+        for j_path, version_content in jboss_version_dict.items():
+            lines = version_content.strip().splitlines()
+            new_context = Context(content=lines, path=j_path)
+            self._versions.append(JbossVersion(new_context))
```

Comment:
See my comment above, this could just be a call to the separate parse function, no need to create `Content` here.

Reply:
Since we are keeping `JBossVersion` parser it would be cleaner to pull out the content parsing so that you don't have to create a fake context wrapper to call the parser here don't you think?  I don't believe we have any other dependencies between parsers like this do we?

Reply:
Updated